File: app/src/main/java/org/xdty/callerinfo/activity/BaseActivity.java
package org.xdty.callerinfo.activity;

import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import org.xdty.callerinfo.utils.Utils;

public abstract class BaseActivity extends AppCompatActivity {
    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        Utils.checkLocale(getApplicationContext());
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Utils.checkLocale(getApplicationContext());

        setContentView(getLayoutId());
        setTitle(getTitleId());

    }

    protected abstract int getLayoutId();

    protected abstract int getTitleId();

}

File: app/src/main/java/org/xdty/callerinfo/activity/BaseActivity.kt
package org.xdty.callerinfo.activity

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import org.xdty.callerinfo.utils.Utils

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutId)
        setTitle(titleId)
    }

    override fun attachBaseContext(newBase: Context) {
        val context: Context = Utils.changeLang(newBase)
        super.attachBaseContext(context)
    }

    protected abstract val layoutId: Int
    protected abstract val titleId: Int
}